docs/DeveloperGuide.md

```diff
@@ -1,38 +1,662 @@
+
+
+![ArchitectureDiagram](![img.png](img/architecture_diagram.png)
+
+The **_Architecture Diagram_** given above explains the high-level design of the PulsePilot.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+`Main` is responsible for the initialising, processing of user input and termination of the bot. It creates a `PulsePilot` instance.
+
+The application can be further broken down into the following packages:
+
+- `Ui`: The user interface of PulsePilot.
+- `Storage`: Contains the data storage components for PulsePilot.
+- `Health`: Stores health-related information.
+- `Workout`: Stores workout-related information.
+- `Utility`: Contains utility functions, such as input parsing and validation.
+- `Constants`: Contains all constants used in PulsePilot.
+
+###### [Back to table of contents](#table-of-contents)
+
+### UI Package
+
+The `UI` package contains `Handler` and `Output`, which are responsible for handling user input and printing of output fo the screen respectively.
+
+![UI Package Class Diagram](img/ui_class_diagram.png)
+
+#### Handler
+
+The main entry point of the application is the `Handler` class, used to handle user input. When the user starts the bot, the steps are as follows:
+
+1. `initialiseBot()` is called, and it prompts the user to enter their name, which is used throughout the bot.
+2. `initialiseScanner()` is used to create the singleton `Scanner` instance to read user input.
+3. When the user enters input, it is passed to `processInput()`, which determines the type of command inputted.
+4. Once the correct input has been determined, it is passed to the relevant handle function. For example, a `workout` command is passed to `handleWorkout()`.
+
+On termination of the bot, `destroyScanner()` is called to close the `Scanner` created, and `terminateBot()` is called to save the data stored within the bot and exit gracefully.
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Output
+
+The `Output` class is responsible for printing messages, prompts, errors and other information to the terminal for the user.
+
+{To include description and `Output` class diagram}
+
+###### [Back to table of contents](#table-of-contents)
+
+### Workout Package
+
+1. `Workout` is a class that stores the date of the workout.
+2. `Run` is a subclass of Workout and stores the distance, time, pace, and date of the run.
+3. `Gym` is a subclass of Workout and stores the date and an array of `GymStation` objects
+4. `GymStation` stores the name of the gym station, number of sets, and an array of `GymSet` objects.
+5. `GymSet` stores the weight and repetitions for a particular set.
+6. `WorkoutList` is a class that stores an array list different `Workout` objects using ArrayList.
+
+![Workout Package Class Diagram](img/workout_class_diagram.png)
```

Comment:
classes should be represented by text such as {abstract} instead of circled letters in the diagram